test/babel-7-8-compat/babel-7-plugins-with-babel-8-core.sh

```diff
@@ -28,11 +30,30 @@ cd ../..
 
 export YARN_ENABLE_IMMUTABLE_INSTALLS=false
 
-# Build and test
+# Build
 yarn constraints --fix
 make -j use-cjs
-yarn jest $(node -p 'require("./test/babel-7-8-compat/data.json")["babel7plugins-babel8core"].join(" ")' || exit)
+
+# Make sure that packages we want to test share all the same hoisted @babel/core,
+# rather than having their own copies. This allows pre-loading it (for the CJS proxy),
+# and matches usage outside of the monorepo (where @babel/core is provided by the user).
+export TOP_LEVEL_CORE_VERSION=$(node -e '
+  const pkg = require("./package.json");
```

Comment:
I'm not using `yarn constraints` here because this needs to happen _after_ `make use-cjs`, which needs to build Babel with the version _currently_ defined in `package.json`.